From 3401690543acf341ecc8ccd54c9d7d2077f5d2a5 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Sat, 06 Feb 2021 14:10:00 +0200
Subject: [PATCH] [SW-20154] make 4B termination platform specific

This is needed in Goya2 only. Since the fix for this
is already present in Gaudi2 RTL, this setting is not
required.

Change-Id: I2d5de173289f95a35fc8ecd99ebc5b6f9d5fe439
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/common/pcie.c b/common/pcie.c
index 1f866e3..eb5e368 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -191,17 +191,8 @@
 
 	/*
 	 * check and enable 4B termination
-	 * set bit[1] of mmPCIE_WRAP_SPECIAL_GLBL_SPARE_0 register
-	 * to enable LBW 4B termination.
 	 */
-	if (arcboot_rom_global.config_fuse &
-				EFUSE_SYSCFG_ENABLE_4B_TERMINATION) {
-		uint32_t spare0;
-
-		spare0 = regmap_readl(mmPCIE_WRAP_SPECIAL_GLBL_SPARE_0);
-		spare0 |= BIT(1);
-		regmap_writel(spare0, mmPCIE_WRAP_SPECIAL_GLBL_SPARE_0);
-	}
+	check_and_enable_4b_termination();
 
 	/* Disable PCIe error responses (both HBW and LBW) */
 	regmap_writel(0x1, mmPCIE_WRAP_PCIE_RSP_OK);
diff --git a/common/security.h b/common/security.h
index 196fb62..bd2d942 100644
--- a/common/security.h
+++ b/common/security.h
@@ -135,5 +135,6 @@
 void security_jtag_rr_hbw_config(uint8_t spi_controller_en);
 void security_jtag_dap(uint8_t status);
 void security_elbi_rr_lbw_config(uint8_t spi_controller_en);
+void check_and_enable_4b_termination(void);
 
 #endif /* _SECURITY_H_ */
diff --git a/gaudi2_arc/security.c b/gaudi2_arc/security.c
index 702cdcb..671db24 100644
--- a/gaudi2_arc/security.c
+++ b/gaudi2_arc/security.c
@@ -158,3 +158,8 @@
 			       SECURITY_EFUSE_KEYS_OFFSET + (hash_key_index *
 			       SECURITY_BYTE_PER_DWORD));
 }
+
+void check_and_enable_4b_termination(void)
+{
+	/* Not needed for Gaudi2 */
+}
diff --git a/goya2_arc/security.c b/goya2_arc/security.c
index 5290a38..4bee9aa 100644
--- a/goya2_arc/security.c
+++ b/goya2_arc/security.c
@@ -388,3 +388,20 @@
 	ret += (key_soc_location_index * sizeof(uint32_t));
 	return ret;
 }
+
+void check_and_enable_4b_termination(void)
+{
+	uint32_t spare0;
+	/*
+	 * check and enable 4B termination
+	 * set bit[1] of mmPCIE_WRAP_SPECIAL_GLBL_SPARE_0 register
+	 * to enable LBW 4B termination.
+	 */
+	if (arcboot_rom_global.config_fuse &
+				EFUSE_SYSCFG_ENABLE_4B_TERMINATION) {
+
+		spare0 = regmap_readl(mmPCIE_WRAP_SPECIAL_GLBL_SPARE_0);
+		spare0 |= BIT(1);
+		regmap_writel(spare0, mmPCIE_WRAP_SPECIAL_GLBL_SPARE_0);
+	}
+}
